src/Shared/SqlProcessor.cs

```diff
@@ -228,177 +680,241 @@ private static bool SanitizeNumericLiteral(string sql, ref int index)
                 break;
             }
 
-            index = i;
+            state.ParsePosition = ++i;
+
+            buffer[state.SanitizedPosition++] = '?';
             return true;
         }
 
         return false;
     }
 
-    private static void WriteToken(string sql, ref int index, SqlProcessorState state)
+    private ref struct ParseState
     {
-        var i = index;
-        var ch = sql[i];
+        // ParseState intentionally uses public fields (not properties):
+        // - This is a ref struct that lives on the stack and is passed by ref through hot paths.
+        // - Fields avoid property accessor calls in tight loops and yield smaller/faster code after inlining.
+        // - Grouping Span<> and larger struct fields first helps layout and may reduce padding.
+        // - Keeping the struct simple and flat minimizes stack pressure and lets the JIT keep values in registers.
 
-        if (LookAhead("SELECT", sql, ref i, state) ||
-            LookAhead("UPDATE", sql, ref i, state) ||
-            LookAhead("INSERT", sql, ref i, state) ||
-            LookAhead("DELETE", sql, ref i, state) ||
-            LookAheadDdl("CREATE", sql, ref i, state) ||
-            LookAheadDdl("ALTER", sql, ref i, state) ||
-            LookAheadDdl("DROP", sql, ref i, state) ||
-            LookAhead("INTO", sql, ref i, state, false, true) ||
-            LookAhead("FROM", sql, ref i, state, false, true, true) ||
-            LookAhead("JOIN", sql, ref i, state, false, true))
-        {
-            i -= 1;
-        }
-        else if (char.IsLetter(ch) || ch == '_')
-        {
-            for (; i < sql.Length; i++)
-            {
-                ch = sql[i];
-                if (char.IsLetter(ch) || ch == '_' || ch == '.' || char.IsDigit(ch))
-                {
-                    state.SanitizedSql.Append(ch);
-                    continue;
-                }
+        // Stored in state to avoid slicing repeatedly.
+        public Span<char> SummaryBuffer; // 16 bytes (on x64)
 
-                break;
-            }
-
-            if (state.CaptureNextTokenAsTarget)
-            {
-                state.CaptureNextTokenAsTarget = false;
-#if NET
-                state.DbQuerySummary.Append(' ').Append(sql.AsSpan(index, i - index));
-#else
-                var collection = sql.Substring(index, i - index);
-                state.DbQuerySummary.Append(' ').Append(collection);
-#endif
-            }
+        public SqlKeywordInfo? PreviousParsedKeyword; // 8 bytes (reference type)
 
-            i -= 1;
+        public SqlKeyword FirstSummaryKeyword; // 4 bytes (enum, underlying int)
+        public SqlKeyword PreviousSummaryKeyword; // 4 bytes (enum, underlying int)
```

Comment:
Is there any advantage to using a smaller type like `short` (or `byte`) for the enum given the number of possible values is relatively small?

Reply:
I can try it and test. As we're passinf by-ref I don't think it should make much difference, if any.

Reply:
Just suggested as the comments suggest field size is important.